### PR TITLE
Code cleaning.

### PR DESCRIPTION
**Pull request by skarasov**
Code cleaning.
CLang adaptation.
Ref iterator in for loop in poligon.cpp.
Removed excessive ";".
Fix ret val of operator-- in LayerIndex.h.

### CODE REVIEW

File: src/layerPart.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ void createLayerParts(SliceMeshStorage& mesh, Slicer* slicer)
 
     // OpenMP compatibility fix for GCC <= 8 and GCC >= 9
     // See https://www.gnu.org/software/gcc/gcc-9/porting_to.html, section "OpenMP data sharing"
-#if defined(__GNUC__) && __GNUC__ <= 8
```
Comment (Ghostkeeper) [inline src/layerPart.cpp:98]: If `__GNUC__` is defined, it should never also have `__clang__` defined, right?

Comment (skarasov) [reply src/layerPart.cpp:98]: Maybe, but Clang 10 with MinGW traps here.

File: libs/rapidjson/rapidjson.h [hunk 2]
```diff
@@ -1,15 +1,15 @@
 // Tencent is pleased to support the open source community by making RapidJSON available.
-//
```
Comment (rburema) [inline libs/rapidjson/rapidjson.h:2]: 'rapidjson' is in the libs folder. The libs folder contains a file saying:
```
cat .\libs\DO_NOT_CHANGE_THESE
These libraries are extern, and while they're utilized by this project,
they are in fact not actually developed on this repository.
If changes are desired for these projects, please refer to their official repositories.
```

**Comment by rburema** [comment]
I merged the appliccable parts of this PR in https://github.com/Ultimaker/CuraEngine/commit/5e6f3d45a2e4cac11f0423e72edc9619392fccde ... github is not registering this as merging this PR, so I'm closing it manually.

**Pull request closed without merging**